From bcb91edc06c9959f1f5bb57a07c963868ab34b69 Mon Sep 17 00:00:00 2001
From: Christophe Priouzeau <christophe.priouzeau@st.com>
Date: Thu, 23 Aug 2018 14:34:51 +0200
Subject: [PATCH 1/2] MAKEFILE: disable igb for arm platform

Signed-off-by: Christophe Priouzeau <christophe.priouzeau@st.com>
---
 Makefile | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/Makefile b/Makefile
index 2a931c3..9e193d5 100644
--- a/Makefile
+++ b/Makefile
@@ -35,18 +35,18 @@ help:
 	@echo '  clean: a summary clean target to clean _all_ folders'
 	@echo ''
 
-igb: FORCE
-	$(call descend,kmod/$@)
+# igb: FORCE
+#         $(call descend,kmod/$@)
 
-igb_clean:
-	$(call descend,kmod/igb/,clean)
+# igb_clean:
+#         $(call descend,kmod/igb/,clean)
 
 lib: FORCE
-	$(call descend,lib/igb)
+	#$(call descend,lib/igb)
 	$(call descend,lib/common)
 
 lib_clean:
-	$(call descend,lib/igb/,clean)
+	#$(call descend,lib/igb/,clean)
 	$(call descend,lib/common/,clean)
 
 mrpd:
@@ -155,8 +155,8 @@ examples_all: examples_common simple_talker simple_listener mrp_client live_stre
 examples_all_clean: examples_common_clean simple_talker_clean simple_listener_clean mrp_client_clean \
 	jackd-talker_clean jackd-listener_clean live_stream_clean simple_rx_clean
 
-all: igb lib daemons_all examples_all avtp_pipeline avtp_avdecc
+all: lib daemons_all examples_all avtp_pipeline avtp_avdecc
 
-clean: igb_clean lib_clean daemons_all_clean examples_all_clean avtp_pipeline_clean avtp_avdecc_clean
+clean: lib_clean daemons_all_clean examples_all_clean avtp_pipeline_clean avtp_avdecc_clean
 
 .PHONY: FORCE
-- 
2.7.4

