From c62f6e2ffb32233bfd285cb7163005edc14957af Mon Sep 17 00:00:00 2001
From: Maxim Uvarov <maxim.uvarov@linaro.org>
Date: Thu, 23 Apr 2020 08:06:51 +0100
Subject: [PATCH 6/6] pta: pseudo: optee3.7 compat

Signed-off-by: Maxim Uvarov <maxim.uvarov@linaro.org>
---
 core/arch/arm/pta/device.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/pta/device.c b/core/arch/arm/pta/device.c
index ae74f898..13c88215 100644
--- a/core/arch/arm/pta/device.c
+++ b/core/arch/arm/pta/device.c
@@ -8,7 +8,6 @@
  * UUIDs which can act as TEE bus devices.
  */
 
-#include <config.h>
 #include <kernel/linker.h>
 #include <kernel/early_ta.h>
 #include <kernel/pseudo_ta.h>
@@ -54,12 +53,14 @@ static TEE_Result get_devices(uint32_t types,
 	SCATTERED_ARRAY_FOREACH(ta, pseudo_tas, struct pseudo_ta_head)
 		add_ta(ta->flags, &ta->uuid, buf, blen, &pos);
 
-	if (IS_ENABLED(CFG_EARLY_TA)) {
+#ifdef	CFG_EARLY_TA
+	{
 		const struct early_ta *eta = NULL;
 
 		for_each_early_ta(eta)
 			add_ta(eta->flags, &eta->uuid, buf, blen, &pos);
 	}
+#endif
 
 	params[0].memref.size = pos;
 	if (pos > blen)
-- 
2.17.1

