From 32f1c9d0e48a1ddc1b804d97d73abb1a490c7255 Mon Sep 17 00:00:00 2001
From: Christophe Priouzeau <christophe.priouzeau@linaro.org>
Date: Wed, 22 Jan 2020 16:27:52 +0100
Subject: [PATCH 1/2] STM32mp157c-DK2

Signed-off-by: Christophe Priouzeau <christophe.priouzeau@linaro.org>
---
 include/configs/stm32mp1.h | 19 +++++++++----------
 1 file changed, 9 insertions(+), 10 deletions(-)

diff --git a/include/configs/stm32mp1.h b/include/configs/stm32mp1.h
index d42a7860be..c294762d55 100644
--- a/include/configs/stm32mp1.h
+++ b/include/configs/stm32mp1.h
@@ -114,16 +114,11 @@
  */
 #define STM32MP_BOOTCMD "bootcmd_stm32mp=" \
 	"echo \"Boot over ${boot_device}${boot_instance}!\";" \
-	"if test ${boot_device} = serial || test ${boot_device} = usb;" \
-	"then stm32prog ${boot_device} ${boot_instance}; " \
-	"else " \
-		"run env_check;" \
-		"if test ${boot_device} = mmc;" \
-		"then env set boot_targets \"mmc${boot_instance}\"; fi;" \
-		"if test ${boot_device} = nand;" \
-		"then env set boot_targets ubifs0; fi;" \
-		"run distro_bootcmd;" \
-	"fi;\0"
+	"env set boot_instance \"0\";" \
+	"env set boot_device \"mmc\";" \
+	"env set boot_targets \"mmc${boot_instance}\";" \
+	"run distro_bootcmd;" \
+	"\0"
 
 #include <config_distro_bootcmd.h>
 
@@ -194,6 +189,10 @@
 	"ramdisk_addr_r=0xc4400000\0" \
 	"fdt_high=0xffffffff\0" \
 	"initrd_high=0xffffffff\0" \
+	"fdtfile=" CONFIG_DEFAULT_DEVICE_TREE ".dtb\0" \
+	"env_default=1\0"				\
+	"env_check=if test $env_default -eq 1;"\
+		" then env set env_default 0;env save;fi\0" \
 	"altbootcmd=run bootcmd\0" \
 	"env_default=1\0" \
 	"env_check=if test $env_default -eq 1;"\
-- 
2.17.1

