From b114e21e9738afc3dec766cedaf10dc243e77cd4 Mon Sep 17 00:00:00 2001
From: Maxim Uvarov <maxim.uvarov@linaro.org>
Date: Tue, 14 Apr 2020 19:48:30 +0100
Subject: [PATCH 07/10] optee add device table

Signed-off-by: Maxim Uvarov <maxim.uvarov@linaro.org>
---
 drivers/tee/optee/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/tee/optee/core.c b/drivers/tee/optee/core.c
index 1854a3db7..e9263967b 100644
--- a/drivers/tee/optee/core.c
+++ b/drivers/tee/optee/core.c
@@ -698,6 +698,8 @@ static const struct of_device_id optee_match[] = {
 	{},
 };
 
+MODULE_DEVICE_TABLE(of, optee_match);
+
 static struct optee *optee_svc;
 
 static int __init optee_driver_init(void)
-- 
2.17.1

