From f8263a1172d502f73f0b4b061c4b43d5070969e0 Mon Sep 17 00:00:00 2001
From: Maxim Uvarov <maxim.uvarov@linaro.org>
Date: Tue, 3 Mar 2020 08:26:23 +0000
Subject: [PATCH] fix position independent code without dtb

Signed-off-by: Maxim Uvarov <maxim.uvarov@linaro.org>
---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index 0af89e0a78..10c43e82f6 100644
--- a/Makefile
+++ b/Makefile
@@ -904,8 +904,10 @@ ALL-y += $(CONFIG_BUILD_TARGET:"%"=%)
 endif
 
 ifdef CONFIG_INIT_SP_RELATIVE
+ifneq ($(CONFIG_DEFAULT_DEVICE_TREE), "")
 ALL-y += init_sp_bss_offset_check
 endif
+endif
 
 ifeq ($(CONFIG_MPC85xx)$(CONFIG_OF_SEPARATE),yy)
 ALL-y += u-boot-with-dtb.bin
-- 
2.17.1

