From 0fa72a2c7c011454887277ee719ecb9f7c7e77af Mon Sep 17 00:00:00 2001
From: Christophe Priouzeau <christophe.priouzeau@st.com>
Date: Tue, 5 Nov 2019 09:26:09 +0100
Subject: [PATCH] MAKEFILE: disable igb for arm platform

Signed-off-by: Christophe Priouzeau <christophe.priouzeau@st.com>
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index f31b002a..e5f974a4 100644
--- a/Makefile
+++ b/Makefile
@@ -42,11 +42,11 @@ atl_lib_clean:
 	$(call descend,lib/common/,clean)
 
 lib: FORCE
-	$(call descend,lib/igb_avb/lib)
+	#$(call descend,lib/igb_avb/lib)
 	$(call descend,lib/common)
 
 lib_clean:
-	$(call descend,lib/igb_avb/lib/,clean)
+	#$(call descend,lib/igb_avb/lib/,clean)
 	$(call descend,lib/common/,clean)
 
 mrpd:
-- 
2.17.1

