From 54c361139abf3c4ef59673027c7c0f71e3bebe0d Mon Sep 17 00:00:00 2001
From: Maxim Uvarov <maxim.uvarov@linaro.org>
Date: Fri, 22 Nov 2019 14:16:44 +0000
Subject: [PATCH] qemuarm64.dts: add ftpm support

Signed-off-by: Maxim Uvarov <maxim.uvarov@linaro.org>
---
 virt.dts | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/virt.dts b/virt.dts
index b9754fc..59d7801 100644
--- a/virt.dts
+++ b/virt.dts
@@ -6,6 +6,12 @@
 	#address-cells = <0x2>;
 	compatible = "linux,dummy-virt";
 
+	tpm@0 {
+		compatible = "microsoft,ftpm";
+		linux,sml-base = <0x0 0xC0000000>;
+		linux,sml-size = <0x10000>;
+	};
+
 	platform@c000000 {
 		interrupt-parent = <0x8001>;
 		ranges = <0x0 0x0 0xc000000 0x2000000>;
-- 
2.17.1

