From d188a3bc7d0556737e7049726304bf4d6ec5553c Mon Sep 17 00:00:00 2001
From: Christophe Priouzeau <christophe.priouzeau@st.com>
Date: Thu, 23 Aug 2018 14:35:13 +0200
Subject: [PATCH 2/2] CMAKE: pcap support through openembedded

Signed-off-by: Christophe Priouzeau <christophe.priouzeau@st.com>
---
 lib/avtp_pipeline/cmake/FindPCAP.cmake | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/avtp_pipeline/cmake/FindPCAP.cmake b/lib/avtp_pipeline/cmake/FindPCAP.cmake
index 2be3fe3..15f62f7 100644
--- a/lib/avtp_pipeline/cmake/FindPCAP.cmake
+++ b/lib/avtp_pipeline/cmake/FindPCAP.cmake
@@ -52,6 +52,8 @@ IF(EXISTS $ENV{PCAPDIR})
     pcap.h
     PATHS
       $ENV{PCAPDIR}
+    PATHS
+      $ENV{PCAPDIR}/include
     NO_DEFAULT_PATH
   )
 
@@ -60,6 +62,8 @@ IF(EXISTS $ENV{PCAPDIR})
       pcap
     PATHS
       $ENV{PCAPDIR}
+    PATHS
+      $ENV{PCAPDIR}/lib
     NO_DEFAULT_PATH
   )
 
-- 
2.7.4

